packages/flutter_tools/lib/src/isolated/native_assets/native_assets.dart

```diff
@@ -410,417 +565,363 @@ Uri nativeAssetsBuildUri(Uri projectUri, OS os) {
   return projectUri.resolve('$buildDir/native_assets/$os/');
 }
 
-class HotRunnerNativeAssetsBuilderImpl implements HotRunnerNativeAssetsBuilder {
-  const HotRunnerNativeAssetsBuilderImpl();
+/// With `flutter run -d all` we need a place to store the native assets
+/// mapping for multiple OSes combined.
+Uri _buildUriMultiple(Uri projectUri) {
+  final String buildDir = build_info.getBuildDirectory();
+  return projectUri.resolve('$buildDir/native_assets/multiple/');
+}
 
-  @override
-  Future<Uri?> dryRun({
-    required Uri projectUri,
-    required FileSystem fileSystem,
-    required List<FlutterDevice> flutterDevices,
-    required String packageConfigPath,
-    required PackageConfig packageConfig,
-    required Logger logger,
-  }) async {
-    final NativeAssetsBuildRunner buildRunner = NativeAssetsBuildRunnerImpl(
-      projectUri,
-      packageConfigPath,
-      packageConfig,
-      fileSystem,
-      globals.logger,
-    );
-    return dryRunNativeAssets(
-      projectUri: projectUri,
-      fileSystem: fileSystem,
-      buildRunner: buildRunner,
-      flutterDevices: flutterDevices,
-    );
+Map<NativeCodeAssetImpl, KernelAsset> _assetTargetLocationsWindowsLinux(
```

Comment:
The previous reviewers preferred not having OS-specific things in this OS-agnostic file. Please rename back to `_assetTargetLocationsSingleArchitecture` to comply with their wishes.

And ditto for code moving.

Reply:
If we have `_assetTargetLocations{Android,IOS,MacOS}` a reader of this code would naturally look for `_assetTargetLocations{Linux,Windows}` and cannot find them - which is the first confusing.

If we name it `_assetTargetLocationsSingleArchitecture`, then one could think that `_assetTargetLocationsAndroid`  means it's always multiple architectures, but it's not necessarily the case.

The difference between these functions isn't whether it's a single architecture or multiple but which operating system they deal with. So I think it makes more sense to have this naming.

@dcharkes If you feel strongly about this, I can rename it back, but I think it's worse for a reader of this code.

Reply:
> @dcharkes If you feel strongly about this, I can rename it back, but I think it's worse for a reader of this code.

I don't, I had `LinuxWindows` in my initial PRs as well before the reviewers on those PRs requested changes.

Reply:
Then I'll leave it as it is.